src/commonMain/kotlin/fr/acinq/eclair/channel/Channel.kt

```diff
@@ -2819,6 +2861,64 @@ data class Closing(
 
         return txs[tx.txid]?.let { (_, desc) -> fee(tx)?.let { it to desc } }
     }
+
+    private fun storeChannelClosed(additionalConfirmedTx: Transaction?): ChannelAction.Storage.StoreChannelClosed {
+        // We want to give the user the list of btc transactions for their outputs
+        val txids = mutableListOf<ByteVector32>()
+        var claimed = 0.sat
+        var type = ChannelClosingType.Other
```

Comment:
The closing type depends on the channel's xxxCommitPublish types, but it's not very clear what types takes precedence.

How about something like:

```kotlin
val type = when {
    mutualClosePublished.isNotEmpty() -> ChannelClosingType.Mutual
    localCommitPublished != null -> ChannelClosingType.Local
    remoteCommitPublished != null -> ChannelClosingType.Remote
    else -> ChannelClosingType.Other // includes revoked commit published case
}
```

Reply:
Done in 3f65583

I think that if `nextRemoteCommitPublished` is non-null, then the type would also be `ChannelClosingType.Remote`. However I'm not 100% sure about `futureRemoteCommitPublished`. Please check.

Reply:
Yes, 3f65583 is correct to me